File: sdk/typescript/examples/mandate-bot.ts
/**
 * Cooperative reference bot: follows welfare mandates over WebSocket.
 * Run with:  node examples/mandate-bot.ts
 */
import { TradingClient, Agent, MandateStrategy } from '../src/index.js'

const url = process.env.EXCHANGE_URL ?? 'http://127.0.0.1:8080'
const name = process.env.AGENT_NAME ?? 'mandate-ts'
const durationMs = Number(process.env.DURATION_MS ?? 90_000)

const client = new TradingClient(url)
const agent = await Agent.create(client, name)
console.log(`agent ${name} = ${agent.agentId}`)
await agent.run(new MandateStrategy(), { durationMs })

const desk = await client.agent(agent.agentId)
console.log(`final equity: $${desk.equity.toFixed(2)} · role: ${desk.role}`)

File: sdk/python/examples/mandate_bot.py
"""Cooperative reference bot driven purely by welfare mandates."""

from trading_engine import Agent, MandateStrategy, TradingClient

client = TradingClient("http://127.0.0.1:8080")
agent = Agent.create(client, "mandate-bot")
agent.run(MandateStrategy(), duration_s=120, log=print)
